add SignKey gpg entity param to allow easier pgp signing of commits

anandkumarpatel opened the pull request:
This PR creates a new param for the `Commit` struct called `SignKey`.
`SignKey` is an `openpgp.Entity` type which is the standard way of reading and validating pgp keys.

When specified the key will be used to sign the commit. This makes it so the user does not have to sign the key externally and duplicate some of the logic the `CreateCommit` function does.

To maintain backward compatibility, if the `Verification.Signature` is defined `SignKey` will have no effect. 

This provides similar functionality that go-git has: https://github.com/src-d/go-git/commit/7b6c1266556f59ac436fada3fa6106d4a84f9b56

Note from the git docs: https://github.com/octokit/routes/blob/152c0b5d156844e17a1805d0509c9a0b7cfc848f/routes/api.github.com/git.json#L282
```
To pass a `signature` parameter, you need to first manually create a valid PGP signature, which can be complicated. You may find it easier to [use the command line]
```
So I wanted to make it less complicated 😄

gmlewis commented:
Thank you, @anandkumarpatel.

In the past, when "helper" functions have been suggested for this repo, we have typically requested that they be put in their own separate repo for a few reasons: it keeps the API surface area of this repo smaller, reduces the maintenance burden on this repo, and keeps its list of external dependencies smaller.

However, I can see how this can be useful, and I'm game to include it in this repo.

I'd like @willnorris and @gauntface to weigh in on their thoughts for including something like this first.

willnorris commented on github/git_commits.go:
```diff
@@ -37,6 +40,11 @@ type Commit struct {
+	// SignKey denotes a key to sign the commit with. If not nil this key will
+	// be used to sign the commit. The private key must be present and already
+	// decrypted. Ignored if Verification.Signature is defined.
+	SignKey *openpgp.Entity
```
nit: probably calling this `SigningKey` ?

Also, add a `json:"-"` tag to explicitly note that this does not part of the GitHub representation of this object.

anandkumarpatel replied:
Sounds good, Updated.

gmlewis commented:
Closing abandoned PR.

anandkumarpatel commented:
@gmlewis Sorry, forgot about this one. 
I rebased and updated my branch based on @willnorris feedback. Is it possible to reopen? or should I make another PR?

gmlewis commented:
Oh, sure... no problem!  I was just trying to do some Spring Cleaning in June.  :smile:

gmlewis commented:
![Screenshot from 2019-06-14 09-40-42](https://user-images.githubusercontent.com/6598971/59513417-ceef5a80-8e88-11e9-8955-bdbba8d833ec.png)

Whups, it's not allowing me to reopen because of a force push. Can you please open another PR with these changes and we can run with that one?

anandkumarpatel commented:
No problem: Continued here: https://github.com/google/go-github/pull/1198
